scipy/sparse/base.py

```diff
@@ -675,23 +675,75 @@ def toarray(self, order=None, out=None):
         """
         return self.tocoo().toarray(order=order, out=out)
 
-    def todok(self):
-        return self.tocoo().todok()
+    # In principle, providing a tocsr method will be sufficient for any
+    # sparse matrix format deriving from spmatrix.
+    def tocsr(copy=False):
```

Comment:
Oh dear. I should add tests for these [conversion functions](https://github.com/musically-ut/scipy-sparse-benchmark/blob/master/Scipy-sparse-benchmark.ipynb).